query-graphs-js/src/querygraph.ts

```diff
@@ -748,6 +798,49 @@ function federateSubgraphs(supergraph: Schema, subgraphs: QueryGraph[]): QueryGr
       }
     );
   }
+
+  /**
+   * Handling progressive overrides here. For each progressive @override
+   * application (with a label), we want to update all relevant edges to the
+   * overridden field across all subgraphs with their respective override
+   * condition (the label and a T/F value). The "from" subgraph will have an
+   * override condition of `false`, whereas all other occurrences (all
+   * pertaining specifically to the @shareable interaction) will have an
+   * override condition of `true`.
+   */
+  const allOverrideApplications = schemas
+    .flatMap(s => federationMetadata(s)?.overrideDirective().applications() ?? []);
+
+  for (const application of allOverrideApplications) {
+    const label = application.arguments().label;
+    if (!label) continue;
+
+    const field = application.parent;
+    assert(field instanceof NamedSchemaElement, () => `@override should have been on a field, got ${field}`)
+    const type = field.parent;
+
+    for (const [i, subgraph] of subgraphs.entries()) {
+      const [vertex, ...unexpectedAdditionalVertices] = subgraph.verticesForType(type.name);
+      assert(vertex && unexpectedAdditionalVertices.length === 0, () => `Subgraph ${i} should have exactly one vertex for type ${type.name}`);
+      const edges = subgraph.outEdges(vertex);
+      for (const edge of edges) {
+        if (
+          edge.transition.kind === "FieldCollection"
+          && edge.transition.definition.name === field.name
+        ) {
+          const head = copyPointers[i].copiedVertex(vertex);
+          const copiedEdge = builder.edge(head, edge.index);
+
+          // TODO: from sachin - should this only be applied in 2 places?
```

Comment:
Still need to address this comment (this `overrideCondition` should be applied in two places, the overriding subgraph and the overridden subgraph).

Reply:
I'm sure this can be cleaned up but it works and I'll take it for now. Will revisit last.
e44e0846